hpx/performance_counters/server/base_performance_counter.hpp

```diff
@@ -19,153 +19,163 @@
-        void finalize() {}
-
-        static components::component_type get_component_type()
-        {
-            return components::get_component_type<wrapping_type>();
-        }
-        static void set_component_type(components::component_type t)
-        {
-            components::set_component_type<wrapping_type>(t);
-        }
-
-        ///////////////////////////////////////////////////////////////////////
-        counter_info get_counter_info_nonvirt() const
-        {
-            return this->get_counter_info();
-        }
-
-        counter_value get_counter_value_nonvirt(bool reset)
-        {
-            return this->get_counter_value(reset);
-        }
-
-        counter_values_array get_counter_values_array_nonvirt(bool reset)
-        {
-            return this->get_counter_values_array(reset);
-        }
-
-        void set_counter_value_nonvirt(counter_value const& info)
-        {
-            this->set_counter_value(info);
-        }
-
-        void reset_counter_value_nonvirt()
-        {
-            this->reset_counter_value();
-        }
-
-        bool start_nonvirt()
-        {
-            return this->start();
-        }
-
-        bool stop_nonvirt()
-        {
-            return this->stop();
-        }
-
-        /// Each of the exposed functions needs to be encapsulated into an action
-        /// type, allowing to generate all required boilerplate code for threads,
-        /// serialization, etc.
-
-        /// The \a get_counter_info_action retrieves a performance counters
-        /// information.
-        HPX_DEFINE_COMPONENT_ACTION(base_performance_counter,
-            get_counter_info_nonvirt, get_counter_info_action);
-
-        /// The \a get_counter_value_action queries the value of a performance
-        /// counter.
-        HPX_DEFINE_COMPONENT_ACTION(base_performance_counter,
-            get_counter_value_nonvirt, get_counter_value_action);
-
-        /// The \a get_counter_value_action queries the value of a performance
-        /// counter.
-        HPX_DEFINE_COMPONENT_ACTION(base_performance_counter,
-            get_counter_values_array_nonvirt, get_counter_values_array_action);
-
-        /// The \a set_counter_value_action
-        HPX_DEFINE_COMPONENT_ACTION(base_performance_counter,
-            set_counter_value_nonvirt, set_counter_value_action);
-
-        /// The \a reset_counter_value_action
-        HPX_DEFINE_COMPONENT_ACTION(base_performance_counter,
-            reset_counter_value_nonvirt, reset_counter_value_action);
-
-        /// The \a start_action
-        HPX_DEFINE_COMPONENT_ACTION(base_performance_counter,
-            start_nonvirt, start_action);
-
-        /// The \a stop_action
-        HPX_DEFINE_COMPONENT_ACTION(base_performance_counter,
-            stop_nonvirt, stop_action);
-
-    protected:
-        hpx::performance_counters::counter_info info_;
-        util::atomic_count invocation_count_;
+        class base_performance_counter
```

Comment:
I have no problem reverting the formatting changes again. I'm not exactly sure what happened here. I use `git clang-format` which should only affect the direct changeset (in that case, a forward declaration was removed). The resulting diff is pretty awful...